src/mlpack/core/data/scaler_methods/standardscaler.hpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file Standardscaler.hpp
+ * @author Jeffin Sam
+ *
+ * StandardScaler class to scale features.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STANDARD_SCALE_HPP
+#define MLPACK_CORE_DATA_STANDARD_SCALE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple Standard Scaler class
+ *
+ * Given an input dataset this class helps you to Standardize features
+ * by removing the mean and scaling to unit variance.
```

Comment:
Just FYI, this is not the same as whitening, but it reminded me of some of the whitening functionality we have, like `WhitenUsingSVD()` in `src/mlpack/core/math/lin_alg.hpp`.  You might consider adding a whitening scaling too (maybe some other time), because that can sometimes be a useful preprocessing step for some algorithms.

http://ufldl.stanford.edu/wiki/index.php/Data_Preprocessing#PCA.2FZCA_Whitening is not exactly the same but it is similar.

Reply:
I will surely add it post gsoc, or at lease after i am done with string utilities. I have added it to my todo list :)

Reply:
Also your take on building binding for this ?

Reply:
> Also your take on building binding for this ?

Had some free time up my sleeve, so wrote one , PTAL :)

Reply:
> I will surely add it post gsoc, or at lease after i am done with string utilities. I have added it to my todo list :)

Ok, So none of my PR had any comments, so had some free time left and thus implemented PCA and ZCA whitening, followed this [article](http://ufldl.stanford.edu/tutorial/unsupervised/PCAWhitening/) to implement that, added a test and also added it to binding.

Reply:
Sounds good, sorry for the slow response on this one.  I'll review the PCA/ZCA implementation now.